### PR TITLE
Introduce PlumaWindowActivatable, PlumaAppAcitvatable, PlumaViewActivatable

### PR DESCRIPTION
**Pull request by mbkma**
Sorry for the large PR, backported a lot of plugin-specific commits from gedit.

### CODE REVIEW

File: pluma/Makefile.am [hunk 1]
```diff
@@ -80,7 +81,9 @@ INST_H_FILES =				\
 	pluma-tab.h 			\
 	pluma-utils.h 			\
 	pluma-view.h 			\
-	pluma-window.h
+	pluma-view-activatable.h 	\
+	pluma-window.h	\
```
Comment (rbuj) [inline pluma/Makefile.am:85]: the backslash is not aligned with the others

File: pluma/pluma-app-activatable.h [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+typedef struct _PlumaAppActivatable           PlumaAppActivatable; /* dummy typedef */
+typedef struct _PlumaAppActivatableInterface  PlumaAppActivatableInterface;
+
+struct _PlumaAppActivatableInterface
+{
+    GTypeInterface g_iface;
+
+    /* Virtual public methods */
+    void    (*activate)     (PlumaAppActivatable *activatable);
+    void    (*deactivate)       (PlumaAppActivatable *activatable);
```
Comment (rbuj) [inline pluma/pluma-app-activatable.h:48]: 3rd column isn't aligned

File: pluma/pluma-app-activatable.h [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+};
+
+/*
+ * Public methods
+ */
+GType    pluma_app_activatable_get_type (void)  G_GNUC_CONST;
+
+void     pluma_app_activatable_activate (PlumaAppActivatable *activatable);
+void     pluma_app_activatable_deactivate   (PlumaAppActivatable *activatable);
```
Comment (rbuj) [inline pluma/pluma-app-activatable.h:57]: same here

File: pluma/pluma-app.c [hunk 4]
```diff
@@ -372,196 +407,190 @@ app_weak_notify (gpointer data,
...
+        }
+
+        if ((state & GDK_WINDOW_STATE_STICKY ) != 0)
+            gtk_window_stick (GTK_WINDOW (window));
+        else
+            gtk_window_unstick (GTK_WINDOW (window));
+    }
+
+    g_signal_connect (window,
+                      "focus_in_event",
```
Comment (rbuj) [inline pluma/pluma-app.c:581]: It's customary to write the name of the signal also on the first line. Test:
```
$ git grep g_signal_connect
```

File: pluma/pluma-app.c [hunk 5]
```diff
@@ -746,24 +775,24 @@ _pluma_app_get_window_in_viewport (PlumaApp  *app,
  * a newly allocated list of #PlumaDocument objects
  */
 GList *
-pluma_app_get_documents	(PlumaApp *app)
+pluma_app_get_documents    (PlumaApp *app)
```
Comment (rbuj) [inline pluma/pluma-app.c:778]: do not add extra blank spaces between the function name and arguments

File: pluma/pluma-window-activatable.h [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+
+typedef struct _PlumaWindowActivatable           PlumaWindowActivatable; /* dummy typedef */
+typedef struct _PlumaWindowActivatableInterface  PlumaWindowActivatableInterface;
+
+struct _PlumaWindowActivatableInterface
+{
+    GTypeInterface g_iface;
+
+    /* Virtual public methods */
+    void    (*activate)     (PlumaWindowActivatable *activatable);
```
Comment (rbuj) [inline pluma/pluma-window-activatable.h:46]: 3rd column is no aligned

File: pluma/pluma-window-activatable.h [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+    void    (*deactivate)       (PlumaWindowActivatable *activatable);
+    void    (*update_state)     (PlumaWindowActivatable *activatable);
+};
+
+/*
+ * Public methods
+ */
+GType    pluma_window_activatable_get_type  (void)  G_GNUC_CONST;
+
+void     pluma_window_activatable_activate  (PlumaWindowActivatable *activatable);
```
Comment (rbuj) [inline pluma/pluma-window-activatable.h:56]: 3rd column is no aligned

File: pluma/pluma-window.c [hunk 8]
```diff
@@ -4811,20 +4774,20 @@ pluma_window_get_tab_from_location (PlumaWindow *window,
  *
  * Return value: (transfer none): the #PlumaMessageBus associated with @window
  */
-PlumaMessageBus	*
+PlumaMessageBus    *
```
Comment (rbuj) [inline pluma/pluma-window.c:4777]: there are extra blank spaces between the type and the pointer sign

File: pluma/pluma-window.c [hunk 9]
```diff
@@ -4610,144 +4573,144 @@ _pluma_window_set_default_location (PlumaWindow *window,
...
+    }
 }
 
 static void
-hide_notebook_tabs_on_fullscreen (GtkNotebook	*notebook,
-				  GParamSpec	*pspec,
-				  PlumaWindow	*window)
+hide_notebook_tabs_on_fullscreen (GtkNotebook    *notebook,
+                                  GParamSpec    *pspec,
+                                  PlumaWindow    *window)
```
Comment (rbuj) [inline pluma/pluma-window.c:4633]: variable names aren't aligned

File: pluma/pluma-window.c [hunk 10]
```diff
@@ -4394,53 +4358,52 @@ pluma_window_close_all_tabs (PlumaWindow *window)
  */
 void
 pluma_window_close_tabs (PlumaWindow *window,
-			 const GList *tabs)
+             const GList *tabs)
```
Comment (rbuj) [inline pluma/pluma-window.c:4361]: indent

File: pluma/pluma-window.c [hunk 11]
```diff
@@ -4222,39 +4188,38 @@ pluma_window_create_tab (PlumaWindow *window,
...
-				  const PlumaEncoding *encoding,
-				  gint                 line_pos,
-				  gboolean             create,
-				  gboolean             jump_to)
+pluma_window_create_tab_from_uri (PlumaWindow     *window,
+                                  const gchar     *uri,
+                                  const PlumaEncoding *encoding,
+                                  gint         line_pos,
+                                  gboolean         create,
+                                  gboolean         jump_to)
```
Comment (rbuj) [inline pluma/pluma-window.c:4196]: variable names aren't aligned

File: pluma/pluma-window.c [hunk 12]
```diff
@@ -1281,2841 +1274,2814 @@ recent_manager_changed (GtkRecentManager *manager,
...
+        gtk_drag_dest_set_target_list (GTK_WIDGET (window), tl);
+        gtk_target_list_unref (tl);
+    }
+
+    gtk_target_list_add_uri_targets (tl, TARGET_URI_LIST);
+
+    /* connect instead of override, so that we can
+     * share the cb code with the view */
+    g_signal_connect (window,
+                      "drag_data_received",
```
Comment (rbuj) [inline pluma/pluma-window.c:4038]: it's customary to write the name of the signal also on the first line, review this file since there are more matches.

File: pluma/pluma-window.c [hunk 13]
```diff
@@ -1281,2841 +1274,2814 @@ recent_manager_changed (GtkRecentManager *manager,
...
+                       window->priv->vpaned,
+                       TRUE,
+                       FALSE);
+
+    pluma_debug_message (DEBUG_WINDOW, "Create pluma notebook");
+    window->priv->notebook = pluma_notebook_new ();
+    add_notebook (window, window->priv->notebook);
+
+    /* side and bottom panels */
+      create_side_panel (window);
```
Comment (rbuj) [inline pluma/pluma-window.c:3991]: indent on the left

File: pluma/pluma-window-activatable.c [hunk 14]
```diff
@@ -0,0 +1,115 @@
...
+ *  (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU Library General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Library General Public License
+ *  along with this program; if not, write to the Free Software
+ *  Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA 02111-1307, USA.
```
Comment (rbuj) [inline pluma/pluma-window-activatable.c:19]: from now on it's recommended to use the version that includes the URL instead of the posting address.

https://www.gnu.org/licenses/gpl-howto.html
> You should have received a copy of the GNU General Public License
> along with Foobar.  If not, see <https://www.gnu.org/licenses/>.

Foobar -> Pluma or Foobar `this program ` if you prefer

File: pluma/pluma-window.c [hunk 15]
```diff
@@ -77,214 +77,214 @@ static gboolean cansave = TRUE;
...
-	gint pane_page;
+    gint pane_page;
 
-	pluma_debug (DEBUG_WINDOW);
+    pluma_debug (DEBUG_WINDOW);
 
-	g_settings_set (window->priv->editor_settings, PLUMA_SETTINGS_WINDOW_SIZE,
-			"(ii)",	window->priv->width, window->priv->height);
+    g_settings_set (window->priv->editor_settings, PLUMA_SETTINGS_WINDOW_SIZE,
+                    "(ii)",    window->priv->width, window->priv->height);
```
Comment (rbuj) [inline pluma/pluma-window.c:134]: extra blank spaces

File: pluma/pluma-window.c [hunk 16]
```diff
@@ -77,214 +77,214 @@ static gboolean cansave = TRUE;
...
-	TARGET_URI_LIST = 100
+    TARGET_URI_LIST = 100
 };
 
 G_DEFINE_TYPE_WITH_PRIVATE (PlumaWindow, pluma_window, GTK_TYPE_WINDOW)
 
-static void	recent_manager_changed	(GtkRecentManager *manager,
-					 PlumaWindow *window);
+static void    recent_manager_changed    (GtkRecentManager *manager,
+                                          PlumaWindow *window);
```
Comment (rbuj) [inline pluma/pluma-window.c:104]: the names of the arguments are not aligned in the function header, there are more matches in this file.

File: plugins/Makefile.am [hunk 17]
```diff
@@ -1,7 +1,9 @@
 DIST_SUBDIRS =		\
-	docinfo
+	docinfo	\
+	filebrowser
 
 SUBDIRS = 		\
-	docinfo
+	docinfo	\
+	filebrowser
```
Comment (rbuj) [inline plugins/Makefile.am]: backslash isn't aligned

File: plugins/filebrowser/pluma-file-browser-plugin.c [hunk 18]
```diff
@@ -185,55 +186,55 @@ pluma_file_browser_plugin_get_property (GObject    *object,
 }
 
 static void
-on_end_loading_cb (PlumaFileBrowserStore      * store,
-                   GtkTreeIter                * iter,
-                   PlumaFileBrowserPluginPrivate * data)
+on_end_loading_cb (PlumaFileBrowserStore         *store,
+                   GtkTreeIter                   *iter,
+                   PlumaFileBrowserPluginPrivate * priv)
```
Comment (rbuj) [inline plugins/filebrowser/pluma-file-browser-plugin.c:191]: there is an extra black space between variable name an pointer sign

File: plugins/filebrowser/pluma-file-browser-plugin.c [hunk 19]
```diff
@@ -79,43 +80,43 @@ static void on_uri_activated_cb          (PlumaFileBrowserWidget * widget,
 static void on_error_cb                  (PlumaFileBrowserWidget * widget,
                                           guint code,
                                           gchar const *message,
-                                          PlumaFileBrowserPluginPrivate * data);
+                                          PlumaFileBrowserPluginPrivate * priv);
```
Comment (rbuj) [inline plugins/filebrowser/pluma-file-browser-plugin.c:83]: you should also comment the refactor of the variable name from `data` to `priv` in the commit body, or you can also make this change in another commit.

File: plugins/Makefile.am [hunk 20]
```diff
@@ -1,29 +1,32 @@
...
+	quickopen	\
+	snippets	\
+	sort		\
+	taglist		\
+	time		\
+	trailsave
+
+if ENABLE_ENCHANT
+SUBDIRS      += spell
+endif
```
Comment (rbuj) [inline plugins/Makefile.am]: this commit should be squashed as it fixes a new issue introduced in a previous commit.

File: pluma/pluma-window.c [hunk 21]
```diff
@@ -4109,8 +4110,10 @@ pluma_window_init (PlumaWindow *window)
 	                  G_CALLBACK (on_extension_removed),
 	                  window);
 
-	/* set visibility of panes.
-	 * This needs to be done after plugins activatation */
+	peas_extension_set_call (window->priv->extensions, "activate");
+
+	 /* set visibility of panes.
+	  This needs to be done after plugins activatation */
```
Comment (rbuj) [inline pluma/pluma-window.c]: this breaks the text alignment

File: pluma/pluma-window.c [hunk 22]
```diff
@@ -4096,9 +4096,10 @@ pluma_window_init (PlumaWindow *window)
 	pluma_debug_message (DEBUG_WINDOW, "Update plugins ui");
 
 	window->priv->extensions = peas_extension_set_new (PEAS_ENGINE (pluma_plugins_engine_get_default ()),
-	                                                   PEAS_TYPE_ACTIVATABLE, "object", window, NULL);
-
-	peas_extension_set_call (window->priv->extensions, "activate");
+                                                       PLUMA_TYPE_WINDOW_ACTIVATABLE,
+                                                       "window",
+                                                       window,
+                                                       NULL);
```
Comment (rbuj) [inline pluma/pluma-window.c:4065]: indent

**Comment by mbkma** [comment]
thanks for your review, I will adopt your suggestions and split this PR to make review easier
